components/datetime/src/input.rs

```diff
@@ -313,31 +312,54 @@ impl TimeZoneInput for ExtractedTimeZoneInput {
     }
 }
 
+pub(crate) enum ExtractedDateTimeInputWeekCalculatorError {
+    Missing(&'static str),
+}
+
 impl ExtractedDateTimeInput {
-    pub(crate) fn week_of_month(&self, calculator: &WeekCalculator) -> Result<WeekOfMonth, Error> {
-        let day_of_month = self
-            .day_of_month()
-            .ok_or(Error::MissingInputField(Some("day_of_month")))?;
-        let iso_weekday = self
-            .iso_weekday()
-            .ok_or(Error::MissingInputField(Some("iso_weekday")))?;
+    pub(crate) fn week_of_month(
+        &self,
+        calculator: &WeekCalculator,
+    ) -> Result<WeekOfMonth, ExtractedDateTimeInputWeekCalculatorError> {
+        let day_of_month =
+            self.day_of_month()
+                .ok_or(ExtractedDateTimeInputWeekCalculatorError::Missing(
+                    "day_of_month",
+                ))?;
+        let iso_weekday =
+            self.iso_weekday()
+                .ok_or(ExtractedDateTimeInputWeekCalculatorError::Missing(
+                    "iso_weekday",
+                ))?;
         Ok(calculator.week_of_month(day_of_month, iso_weekday))
     }
 
     pub(crate) fn week_of_year(
         &self,
         calculator: &WeekCalculator,
-    ) -> Result<(FormattableYear, WeekOfYear), Error> {
-        let day_of_year_info = self
-            .day_of_year_info()
-            .ok_or(Error::MissingInputField(Some("day_of_year_info")))?;
-        let iso_weekday = self
-            .iso_weekday()
-            .ok_or(Error::MissingInputField(Some("iso_weekday")))?;
-        let week_of = calculator.week_of_year(day_of_year_info, iso_weekday)?;
+    ) -> Result<(FormattableYear, WeekOfYear), ExtractedDateTimeInputWeekCalculatorError> {
+        let day_of_year_info =
+            self.day_of_year_info()
+                .ok_or(ExtractedDateTimeInputWeekCalculatorError::Missing(
+                    "day_of_year_info",
+                ))?;
+        let iso_weekday =
+            self.iso_weekday()
+                .ok_or(ExtractedDateTimeInputWeekCalculatorError::Missing(
+                    "iso_weekday",
+                ))?;
+        // We don't have any calendars with < 14 days per year, and it's unlikely we'll add one
+        debug_assert!(day_of_year_info.day_of_year >= icu_calendar::week::MIN_UNIT_DAYS);
+        debug_assert!(day_of_year_info.days_in_prev_year >= icu_calendar::week::MIN_UNIT_DAYS);
```

Comment:
Are these assertions new, or did they come from somewhere else?

#4986

Reply:
They protect this result case, and yes, it should have been `days_in_year`: https://github.com/robertbastian/icu4x/blob/7f4411c81d06e84fc915177240736507505f4fb4/components/calendar/src/week_of.rs#L240